app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php

```diff
@@ -27,37 +28,59 @@ class GetStockItemData implements GetStockItemDataInterface
      * @var StockIndexTableNameResolverInterface
      */
     private $stockIndexTableNameResolver;
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
 
     /**
      * @param ResourceConnection $resource
      * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param DefaultStockProviderInterface $defaultStockProvider
      */
     public function __construct(
         ResourceConnection $resource,
-        StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        DefaultStockProviderInterface $defaultStockProvider
     ) {
         $this->resource = $resource;
         $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->defaultStockProvider = $defaultStockProvider;
     }
 
     /**
      * @inheritdoc
      */
     public function execute(string $sku, int $stockId): ?array
     {
-        $stockItemTableName = $this->stockIndexTableNameResolver->execute($stockId);
+        $isDefaultStock = $stockId === $this->defaultStockProvider->getId();
+        if ($isDefaultStock) {
+            $stockItemTableName = $this->resource->getTableName('cataloginventory_stock_status');
+            $qtyColumnName = 'qty';
+            $isSalableColumnName = 'stock_status';
+        } else {
```

Comment:
Can we optimise those if/else statement? For example define in advance and move them away from conditions and change only if required
$qtyColumnName = IndexStructure::QUANTITY;
$isSalableColumnName = IndexStructure::IS_SALABLE;